x/emissions/keeper/inference_synthesis/network_regrets.go

```diff
@@ -394,40 +406,53 @@ func GetCalcSetNetworkRegrets(args GetCalcSetNetworkRegretsArgs) error {
 		if err != nil {
 			return errorsmod.Wrapf(err, "Error updating topic initial regret")
 		}
-
-		// For batch event emission
-		emissions.EmitNewTopicInitialRegretSetEvent(args.Ctx, args.TopicId, blockHeight, updatedTopicInitialRegret)
 	}
 
 	return nil
 }
 
 // Calculate the initial regret for all new workers in the topic
-// denominator = std(regrets[i-1, :]) + epsilon
-// offset = cnorm - 8.25 / pnorm
-// dummy_regret[i] = np.percentile(regrets[i-1, :], 25.) + offset * denominator
-// It is assumed that the regrets are filtered by experience for each actor
-// i.e. if they have not been included in the topic for enough epochs, their regret is ignored.
+// When using experienced workers' regrets:
+//
+//	denominator = std(regrets) + epsilon
+//	offset = cnorm - 8.25 / pnorm
+//	initialRegret = percentile(regrets, 25) + offset * denominator
+
+// When using fallback regrets (not enough experienced workers / cold start):
+//
+//	initialRegret = percentile(regrets, 25)
 func CalcTopicInitialRegret(
 	regrets []alloraMath.Dec,
 	epsilon alloraMath.Dec,
 	pNorm alloraMath.Dec,
 	cNorm alloraMath.Dec,
 	quantileRegret alloraMath.Dec,
 	pNormDiv alloraMath.Dec,
+	usingFallbackRegrets bool,
 ) (initialRegret alloraMath.Dec, err error) {
-	// Calculate the Denominator
+	quantile, err := alloraMath.GetQuantileOfDecs(regrets, quantileRegret)
+	if err != nil {
+		return alloraMath.ZeroDec(), err
+	}
+
+	// If using fallback regrets, just return the quantile
+	if usingFallbackRegrets {
+		return quantile, nil
+	}
```

Comment:
Don't like the idea of splitting up control flow logic between scopes. Would be cleaner if you were to
1. Decide on which scope to do the control flow. In this case, I say the high-level scope (which calls `CalcTopicInitialRegret()`)
2. Pass in just `quantile` instead of `quantileRegret` and `usingFallbackRegrets`. This has impacts in L393 as of this commit.